src/unix/atomic-ops.h

```diff
@@ -42,6 +42,10 @@ UV_UNUSED(static int cmpxchgi(int* ptr, int oldval, int newval)) {
   const int out = (*(volatile int*) ptr);
   __compare_and_swap(ptr, &oldval, newval);
   return out;
+#elif defined(__MVS__)
+  const int out = (*(volatile int*) ptr);
+  cs((cs_t*)&oldval, (cs_t*)ptr, *(cs_t*)(&newval));
```

Comment:
It still isn't fully atomic because there is a time window between the assignment to `out` and the compare-and-swap.  I guess it's acceptable for now because AIX  has the same bug but it should be fixed someday.